activerecord/lib/active_record/core.rb

```diff
@@ -728,8 +729,17 @@ def strict_loading?
     #   user.strict_loading!(false) # => false
     #   user.comments
     #   => #<ActiveRecord::Associations::CollectionProxy>
-    def strict_loading!(value = true)
+    def strict_loading!(value = true, mode: :all)
```

Comment:
If possible we should document the new option here as well with mention to default mode as `:all` and accepted options list.